diagnostic_tool/kinematics_cpp/kinematics.cpp

```diff
@@ -0,0 +1,443 @@
+/*
+ * Adapted from https://github.com/Le0nX/ModernRoboticsCpp
+ */
+#include <cmath>
+#include <vector>
+#include <Eigen/Dense>
+#include <pybind11/pybind11.h>
+#include <pybind11/eigen.h>
+
+namespace py = pybind11;
+
+#define M_PI           3.14159265358979323846  /* pi */
+
+namespace mr {
+
+/* Function: Find if the value is negligible enough to consider 0
+ * Inputs: value to be checked as a double
+ * Returns: Boolean of true-ignore or false-can't ignore
+ */
+    bool NearZero(const double val) {
+        return (std::abs(val) < .000001);
+    }
+
+/* Function: Returns the skew symmetric matrix representation of an angular velocity vector
+ * Input: Eigen::Vector3d 3x1 angular velocity vector
+ * Returns: Eigen::MatrixXd 3x3 skew symmetric matrix
+ */
+    Eigen::Matrix3d VecToso3(const Eigen::Vector3d &omg) {

```

Comment:
Nit: `s/so3/SO3/`?

B.t.w, What is `omg` short for? I am pretty sure it is not for "Oh My God".

Reply:
Kept as is as much of the code is 3rd party source code.